src/test/java/TestLIC4.java

```diff
@@ -0,0 +1,72 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestLIC4 {
+
+    @Test
+    public void testPositive() {
+        Parameters params = new Parameters(0, 0, 0, 0,
+                0, 0, 0, 0, 3, 2, 0, 0,
+                0, 0, 0, 0, 0, 0, 0);
+
+        Point a = new Point(1, 1); //quadrant 1
+        Point b = new Point(2, 2); //quadrant 1
+        Point c = new Point(-1, 10); //quadrant 2
+        Point d = new Point(-4, -3); //quadrant 3
+
+        Point[] pts = {a, b, c, d};
+        Assert.assertEquals(true, Decide.LIC4(pts, params));
+    }
+
+    @Test
+    public void testNegative() {
+        Parameters params = new Parameters(0, 0, 0, 0,
+                0, 0, 0, 0, 2, 2, 0, 0,
+                0, 0, 0, 0, 0, 0, 0);
+        Point a = new Point(1, 1); //quadrant 1
+        Point b = new Point(2, 2); //quadrant 1
+        Point c = new Point(-1, 10); //quadrant 2
+
+        Point[] pts = {a, b, c};
+        Assert.assertEquals(false, Decide.LIC4(pts, params));
+    }
+
+    @Test
+    public void testOnAxis() {
+        Parameters params = new Parameters(0, 0, 0, 0,
+                0, 0, 0, 0, 3, 2, 0, 0,
+                0, 0, 0, 0, 0, 0, 0);
+        Point a = new Point(0, 0); //quadrant 1
+        Point b = new Point(0, -1); //quadrant 3
+        Point c = new Point(-1, 0); //quadrant 2
+
+
+        Point[] pts = {a, b, c};
+        Assert.assertEquals(true, Decide.LIC4(pts, params));
+    }
+
+    @Test
+    public void testIllegalArgument() {
```

Comment:
For testing exceptions JUnit supports declaring that in the annotations, [see junit docs](https://junit.org/junit4/javadoc/4.12/org/junit/Test.html). What approach do we think is best? It might simplify the test, but we might not be able to assert that the error message is correct.

Reply:
That's an interesting question, I think that it would be better to check whether the correct exception message was thrown, but we should discuss this further with the rest of the group and see what they think.

Reply:
I think that we shouldn't hold back PR's now, but discuss it tomorrow. If we decide to do something else, we create an issue and change it.